src/plugins/browser-console.ts

```diff
@@ -83,10 +85,33 @@ export class BrowserConsole {
     this.driver.page.on('pageerror', this.pageErrorEventListener);
   }
 
-  stop() {
+  stop(status?: StatusValue) {
     this.driver.context.off('console', this.consoleEventListener);
     this.driver.page.off('pageerror', this.pageErrorEventListener);
     log(`Plugins: stopped collecting console events`);
-    return this.messages;
+
+    return this.filterOutput(status);
+  }
+
+  filterOutput(status?: StatusValue) {
+    if (status === 'failed') {
+      return this.messages;
+    }
+    if (status === 'succeeded') {
+      // we collect last 10 error, last 5 logs, infos and last 5 warnings and return those
```

Comment:
it would be better to give more priority to errors/warnings and than to log when we know they are present. 

The algorithm could be something like
- Set a Max of 50/100 messages for a journey
- collect all errors, warnings, then get logs if applicable. 

Right now, this would yield different results for every run as the errors/warn/log could happen at different times.